packages/twenty-front/src/modules/workflow/components/WorkflowEditActionFormSendEmail.tsx

```diff
@@ -62,48 +107,115 @@ export const WorkflowEditActionFormSendEmail = ({
 
   const handleSave = form.handleSubmit(saveAction);
 
+  const filter: { or: object[] } = {
+    or: [
+      {
+        accountOwnerId: {
+          eq: currentWorkspaceMember?.id,
+        },
+      },
+    ],
+  };
+
+  if (
+    isDefined(action.settings.connectedAccountId) &&
+    action.settings.connectedAccountId !== ''
+  ) {
+    filter.or.push({
+      id: {
+        eq: action.settings.connectedAccountId,
+      },
+    });
+  }
```

Comment:
could we extract this logic, as long as the select component? I feel this kind of app connection will happen often. It should be re-usable and not specific to `SendEmailAction`. Wdyt?

Reply:
Think we should wait for the next action to be implemented to factorize this logic